exporters/stdout/stdoutlog/exporter.go

```diff
@@ -33,24 +55,115 @@ func New(options ...Option) (*Exporter, error) {
 		timestamps: cfg.Timestamps,
 	}
 	e.encoder.Store(enc)
+	e.selfObservability = newSelfObservability()
 
 	return &e, nil
 }
 
+func newSelfObservability() *selfObservability {
+	if !x.SelfObservability.Enabled() {
+		return nil
+	}
+
+	selfObservability := &selfObservability{
+		enabled: true,
+		attrs: []attribute.KeyValue{
+			semconv.OTelComponentName(fmt.Sprintf("%s/%d", otelComponentType, nextExporterID())),
+			semconv.OTelComponentTypeKey.String(otelComponentType),
+		},
+	}
+
+	mp := otel.GetMeterProvider()
+	m := mp.Meter("go.opentelemetry.io/otel/exporters/stdout/stdoutlog",
+		metric.WithInstrumentationVersion(sdk.Version()),
+		metric.WithSchemaURL(semconv.SchemaURL),
+	)
+
+	var err error
+	if selfObservability.inflightMetric, err = otelconv.NewSDKExporterLogInflight(m); err != nil {
+		otel.Handle(err)
+	}
+	if selfObservability.exportedMetric, err = otelconv.NewSDKExporterLogExported(m); err != nil {
+		otel.Handle(err)
+	}
+	if selfObservability.operationDurationMetric, err = otelconv.NewSDKExporterOperationDuration(m); err != nil {
+		otel.Handle(err)
+	}
+
+	return selfObservability
+}
+
 // Export exports log records to writer.
 func (e *Exporter) Export(ctx context.Context, records []log.Record) error {
 	enc := e.encoder.Load()
 	if enc == nil {
 		return nil
 	}
 
+	var err error
+	if e.selfObservability != nil && e.selfObservability.enabled {
+		err = e.exportWithSelfObservability(ctx, records)
+	} else {
+		err = e.exportWithoutSelfObservability(ctx, records)
+	}
+	return err
+}
+
+const bufferSize = 1024
```

Comment:
Why is this buffer size `1024`? It seems like there are only ever `3` attributes, right?